File: shared/ui/ipc/webview.protocol.notifications.ts
import { NotificationType } from "vscode-jsonrpc";
import { IpcRoutes, WebviewContext } from "./webview.protocol.common";

export interface WebviewDidInitializeNotification {}

export const WebviewDidInitializeNotificationType = new NotificationType<
	WebviewDidInitializeNotification,
	void
>(`${IpcRoutes.Host}/didInitialize`);

export interface WebviewDidChangeContextNotification {
	context: WebviewContext;
}
export const WebviewDidChangeContextNotificationType = new NotificationType<
	WebviewDidChangeContextNotification,
	void
>(`${IpcRoutes.Host}/context/didChange`);

File: vs/src/CodeStream.VisualStudio/Models/Webview.Protocol.Notifications.cs
namespace CodeStream.VisualStudio.Models
{
    public class WebviewDidInitializeNotificationType : NotificationType<EmptyRequestTypeParams>
    {
        public const string MethodName = "host/didInitialize";
        public override string Method => MethodName;
    }

    public class WebviewDidChangeContextNotification
    {
        public WebviewContext Context { get; set; }
    }

    public class WebviewDidChangeContextNotificationType : NotificationType<WebviewDidChangeContextNotification>
    {
        public const string MethodName = "host/context/didChange";
        public override string Method => MethodName;
    }
}
